corehq/messaging/scheduling/forms.py

```diff
@@ -323,13 +352,29 @@ def add_initial_for_timed_schedule(self, initial):
 
         self.add_initial_for_send_time(initial)
 
-        if self.initial_schedule.total_iterations == TimedSchedule.REPEAT_INDEFINITELY:
-            initial['stop_type'] = self.STOP_NEVER
-        elif self.initial_schedule.total_iterations == 1:
-            initial['stop_type'] = self.STOP_AFTER_FIRST_OCCURRENCE
+        if self.initial_schedule.total_iterations == 1:
+            initial['repeat'] = self.REPEAT_NO
         else:
-            initial['stop_type'] = self.STOP_AFTER_OCCURRENCES
-            initial['occurrences'] = self.initial_schedule.total_iterations
+            if initial['send_frequency'] == self.SEND_DAILY:
+                repeat_every = self.initial_schedule.repeat_every
+            elif initial['send_frequency'] == self.SEND_WEEKLY:
+                repeat_every = self.initial_schedule.repeat_every // 7
```

Comment:
Guessing that `self.initial_schedule.repeat_every` is a number of days, and this is setting the local `repeat_every` to the number of weeks. Let me know if that's wrong. It's a little confusing that they both have the name `repeat_every` but different units.

Presumably this conversion is being done to be displayed in the UI. Where is the reverse of this done (converting the value set in the UI back to a number of days)? Is it in [timed_schedule.py](https://github.com/dimagi/commcare-hq/blob/1fbb925274b20c85e42e40e5b6d426855b1c1457/corehq/messaging/scheduling/models/timed_schedule.py#L357)?

Reply:
yeah see the comment by `repeat_every` on the `TimedSchedule` for its meaning in the model. in the ui, this value is always on a scale from 1 to n, incrementing by 1, for all types of schedules (daily, weekly, monthly).

the conversion from the ui value to the model value happens in the `set_simple_weekly_schedule` and `set_simple_monthly_schedule` functions. see the docstrings in those functions for more info